plugins/pay.c

```diff
@@ -454,19 +454,38 @@ static const jsmntok_t *find_worst_channel(const char *buf,
 					   const char *fieldname,
 					   u64 final)
 {
-	u64 prev = final, worstval = 0;
-	const jsmntok_t *worst = NULL, *t;
+	u64 prev, worstval = 0;
+	const jsmntok_t *worst = NULL, *t, *t_prev = NULL;
 	size_t i;
 
 	json_for_each_arr(i, t, route) {
 		u64 val;
+		/* If there's only 2 hops, the worst channel must be the first one. */
+		if (route->size <= 2) {
+			worst = t;
+			break;
+		}
 
 		json_to_u64(buf, json_get_member(buf, t, fieldname), &val);
-		if (worst == NULL || val - prev > worstval) {
-			worst = t;
-			worstval = val - prev;
+
+		/* For the first hop, now we can't know if it's the worst.
+		 * Just store the info and continue. */
+		if (!i) {
+			prev = val;
+			t_prev = t;
+			continue;
+		}
+
+		/* the final hop */
+		if (i + 1 == route->size)
+			assert(val == final);
```

Comment:
We should just drop the final arg, it's now redundant.

Reply:
Yes, I remove this, and also remove the `final` parameter in this interface.